packages/@aws-cdk/aws-codebuild/test/test.project.ts

```diff
@@ -797,6 +798,52 @@ export = {
 
       test.done();
     },
+
+    'certificate arn'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const bucket = new s3.Bucket(stack, 'Bucket');
+
+      // WHEN
+      new codebuild.Project(stack, 'Project', {
+        source: codebuild.Source.s3({
+          bucket,
+          path: 'path',
+        }),
+        environment: {
+          certificate: {
+            bucket,
+            objectKey: 'path',
+          },
+        },
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::CodeBuild::Project', {
+        Environment: objectLike({
+          Certificate: {
+            'Fn::Join': [
+              '',
+              [
+                'arn:',
+                {
+                  'Ref': 'AWS::Partition',
+                },
+                ':s3:::',
+                {
+                  'Ref': stack.getLogicalId(bucket.node.defaultChild as CfnBucket),
+                },
+                '/path',
+              ],
+            ],
```

Comment:
Let's conserve the lines here:

```suggestion
            'Fn::Join': ['', [
              'arn:',
              { 'Ref': 'AWS::Partition' },
              ':s3:::',
              { 'Ref': stack.getLogicalId(bucket.node.defaultChild as CfnBucket) },
              '/path',
            ]],
```